tests/fsharp/core/comprehensions/test.fsx

```diff
@@ -1475,6 +1475,49 @@ module EnumPatternWithFunkyTypes_FSharp_1_0_13904 =
     // This is allowed - 'a is known to be "bool"
     let s = seq { for i in T true -> i }
 
+
+module SideEffectListMonad =
+    type SideEffectListWithReturnBuilder(onReturn, onZero) =
+        member b.Bind(x:unit,f) :list<'b> = f()
+        member b.Combine(x:list<'a>,y:list<'a>) :list<'a> = List.append x y
+        member b.Delay(f:unit->list<'a>) :list<'a> = f()
+        member b.Return _ :list<'a> = onReturn(); []
+        member b.Zero() :list<'a> = onZero(); []
+        member b.Yield(x:'a) :list<'a> = [x]
+
+    let sideEffectListWithReturn onReturn onZero = SideEffectListWithReturnBuilder(onReturn, onZero)
+
+    type SideEffectListWithZeroBuilder(onZero) =
+        member b.Bind(x:unit,f) :list<'b> = f()
+        member b.Combine(x:list<'a>,y:list<'a>) :list<'a> = List.append x y
+        member b.Delay(f:unit->list<'a>) :list<'a> = f()
+        member b.Zero() :list<'a> = onZero(); []
+        member b.Yield(x:'a) :list<'a> = [x]
+
+    let sideEffectListWithZero onZero = SideEffectListWithZeroBuilder(onZero)
+
+    module SideEffectListTests =
+        #if Portable
+        let printfn s = printfn "%s" s
```

Comment:
I know you copy/paste from near tests (good), but maybe this is good opportunity to cleanup this duplication, adding the `printfn` function at top level